install-pcf/azure/README.md

```diff
@@ -30,70 +30,89 @@ the `Contributor` Role on the target Azure Project.
 
    Set your Subscription ID to the subscription that will be used by the install-pcf pipeline:
 
-   ```
+   ```shell
    export SUBSCRIPTION_ID=<YOUR-SUBSCRIPTION-ID>
-   export SERVICE_PRINCIPAL_PASSWORD=<SOME-PASSWORD>
    ```
 
+   ```shell
+   az ad sp create-for-rbac --name "pcfserviceaccount" --role="Contributor" \
+   --scopes="/subscriptions/${SUBSCRIPTION_ID}" | tee app_create.json
    ```
-   az ad app create --display-name "PCFServiceAccount" \
-     --homepage "http://pcfserviceaccount" \
-     --identifier-uris "http://pcfserviceaccount" \
-     --password "$SERVICE_PRINCIPAL_PASSWORD" | tee app_create.json
-
-   export APP_ID="$(jq -r .appId app_create.json)"
 
-   az ad sp create --id "$APP_ID"
-
-   az role assignment create --assignee "http://pcfserviceaccount" \
-     --role "Contributor" \
-     --scope "/subscriptions/$SUBSCRIPTION_ID"
-   ```
+   > Note: The service principal password will output on the console as well as saved in the `app_create.json` file.
 
 2. Create an Azure Storage Account and Container to store terraform.tfstate
 
-   ```
-   az group create --name "pcfci" \
-     --location "WestUS"
+   Set an azure storage account name which is globally unique per datacenter region and change the desired datacenter location.
+  
+    ```shell
+    export STORAGE_NAME="pcfterraform"
+    export REGION="WestUS2"
+    ```
 
-   az storage account create --name "pcfci" \
-     --resource-group "pcfci" \
-     --location "WestUS" \
-     --sku "Standard_LRS"
+    ```shell
+    az group create --name "pcfci" \
+        --location $REGION
 
-   AZURE_STORAGE_ACCOUNT_KEY=$(az storage account keys list --account-name pcfci --resource-group pcfci | jq -r .[0].value)
+    az storage account create --name $STORAGE_NAME \
+        --resource-group "pcfci" \
+        --kind "StorageV2" \
+        --sku "Standard_LRS"
 
-   az storage container create --name terraformstate \
-     --account-name pcfci
-   ```
+    AZURE_STORAGE_ACCOUNT_KEY=$(az storage account keys list --account-name $STORAGE_NAME --resource-group pcfci | jq -r .[0].value)
 
-3. Download `pcf-pipelines` from [Pivotal Networks](https://network.pivotal.io/products/pcf-automation)
+    az storage container create --name terraformstate \
+    --account-name $STORAGE_NAME
```

Comment:
I suggest using the connection string instead of storage account key. Reasons: connection string has a higher priority. If you have an environment variable for the connection string, the account key won't take effect.
For example,
```
connection_string=$(az storage account show-connection-string --resource-group pcfci -n pcfci | jq -r .connectionString)

az storage container create --name terraformstate \
  --connection-string $connection_string
```

Reply:
@dtzar Thanks for quick response.
In my case, I specified an environment variable `AZURE_STORAGE_CONNECTION_STRING` for another storage account (e.g. the account name is `foo`).
If I run `az storage container create --name terraformstate --account-name bar` without specifying `--connection-string`, then the container won't be created in the storage account `bar`. Instead, it's created in `foo`.

And the line `AZURE_STORAGE_ACCOUNT_KEY=$(az storage account keys list --account-name $STORAGE_NAME --resource-group pcfci | jq -r .[0].value)` is not needed.

Reply:
The existing terraform code will create the storage containers in the proper account.